x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -0,0 +1,157 @@
+package keeper
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	evidencetypes "github.com/cosmos/cosmos-sdk/x/evidence/types"
+	"github.com/cosmos/ibc-go/v4/modules/core/exported"
+	ibctmtypes "github.com/cosmos/ibc-go/v4/modules/light-clients/07-tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// HandleConsumerMisbehaviour checks whether the given IBC misbehaviour is valid and, if they are, the misbehaving
+// CheckConsumerMisbehaviour check that the given IBC misbehaviour headers forms a valid light client attack evidence.
+// proceed to the jailing and tombstoning of the bzyantine validators.
+func (k Keeper) HandleConsumerMisbehaviour(ctx sdk.Context, misbehaviour exported.Misbehaviour) error {
```

Comment:
If we want to make this generic, this should be moved to a new `tendermint/misbehaviour.go` since this is tendermint specific. Also a few other functions in this file.